src/main/java/ch/vorburger/exec/ManagedProcessBuilder.java

```diff
@@ -57,9 +60,19 @@
     protected boolean destroyOnShutdown = true;
     protected int consoleBufferMaxLines = 100;
     protected OutputStreamLogDispatcher outputStreamLogDispatcher = new OutputStreamLogDispatcher();
+    protected ManagedProcessListener listener;
     protected List<OutputStream> stdOuts = new ArrayList<OutputStream>();
     protected List<OutputStream> stdErrs = new ArrayList<OutputStream>();
 
+    public ManagedProcessListener getProcessListener() {
+        return listener;
+    }
+
+    public ManagedProcessBuilder setProcessListener(ManagedProcessListener listener) {
+        this.listener= listener;
```

Comment:
ava.lang.Objects#requireNonNull is a 1.7+ java. project set to 1.6, so have to bump it to at least 1.7 for this change. (i'm going to go to 1.8 since we should be moving to that at least anyway)